src/main/java/uk/nhs/digital/nhsconnect/nhais/translator/amendment/mappers/AmendmentAddressToEdifactMapper.java

```diff
@@ -0,0 +1,133 @@
+            .map(AmendmentPatch::getFormattedSimpleValue)
+            .orElse(null);
+        var postalCode = patches.getPostalCode()
+            .map(AmendmentPatch::getFormattedSimpleValue)
+            .orElse(null);
+
+        return PersonAddress.builder()
+            .addressLine1(houseName)
+            .addressLine2(numberOrRoadName)
+            .addressLine3(locality)
+            .addressLine4(postalTown)
+            .addressLine5(county)
+            .postalCode(postalCode)
+            .build();
+    }
+
+    @Override
+    boolean shouldCreateSegment(AmendmentBody amendmentBody) {
+        var patches = amendmentBody.getJsonPatches();
+        return Stream.of(
+            patches.getHouseName(),
+            patches.getNumberOrRoadName(),
+            patches.getLocality(),
+            patches.getPostTown(),
+            patches.getCounty(),
+            patches.getPostalCode())
+            .anyMatch(Optional::isPresent);
+    }
+
+    @Override
+    protected void validatePatches(JsonPatches patches) {
+        validatePostalCodePatchIfExists(patches);
+        checkIfThereAreAllFiveAddressLinesPatches(patches);
+        checkNoPostTownPatchForRemoveOperation(patches);
+        checkLocalityPostTownAndCountyAllEmptyBlankOrAllNotEmptyBlank(patches);
+    }
+
+    private void validatePostalCodePatchIfExists(JsonPatches patches) {
+        if (patches.getPostalCode().isPresent()) {
+            validateNonEmptyValues(Collections.singletonList(patches.getPostalCode()));
+        }
+    }
+
+    private void checkIfThereAreAllFiveAddressLinesPatches(JsonPatches patches) {
+        if(!Stream.of(
+            patches.getHouseName(),
+            patches.getNumberOrRoadName(),
+            patches.getLocality(),
+            patches.getPostTown(),
+            patches.getCounty())
+            .allMatch(Optional::isPresent)) {
+            throw new FhirValidationException("All five address lines must be provided for amendment");
+        }
+    }
+
+    private void checkLocalityPostTownAndCountyAllEmptyBlankOrAllNotEmptyBlank(JsonPatches patches) {
+        if(hasLocalityOrCountyRemovalOperation(patches)) {
+            return;
+        }
+        var localityString = patches.getLocality().get().getValue().get();
+        var countyString = patches.getCounty().get().getValue().get();
+        var postTownString = patches.getPostTown().get().getValue().get();
+        if (!allStringsAreEmptyOrBlank(localityString, countyString, postTownString)
+            && !allStringsAreNotEmptyNorBlank(localityString, countyString, postTownString)) {
+            throw new FhirValidationException("If at least one of the Address - Locality, Address - Post Town and Address County " +
+                "fields is amended for a patient, then the values held for all three of these fields MUST be provided. Actual state: " +
+                "Locality: " + localityString + ", Post Town: " + postTownString + ", County: " + countyString);
+        }
```

Comment:
@mattd-kainos, what do you think about the logic from documentation that you've mentioned today morning? It is checking consistency of locality, county and post town for operations other than removal. Or in case of removal of locality and county should we check if post town is add or replace with non empty value?